### PR TITLE
remove runN if symlink target is gone

### LINKED ISSUE #4323: cylc clean needs to update runN symlink
Say you have
```
~/cylc-run/myflow
|-- run1
|-- run2
`-- runN -> run2
```

`cylc clean myflow/run2` doesn't currently change the target of `runN` to `run1`, instead it will leave it broken.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #4323 

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Review by kinow — COMMENTED** [review]
One test still failing @wxtim 

```bash
=========================== short test summary info ============================
FAILED tests/unit/test_workflow_files.py::test_delete_runN[expect1-dirs1] - A...
====== 1 failed, 963 passed, 1 skipped, 1 xfailed, 12 warnings in 36.87s =======
```

**Review by kinow — APPROVED** [review]

File: tests/unit/test_workflow_files.py [hunk 1]
```diff
@@ -1523,3 +1524,24 @@ def test_get_rsync_rund_cmd(tmp_run_dir: Callable):
...
         'blah/', f'{cylc_run_dir}/']
+
+
+@pytest.mark.parametrize(
+    'expect, dirs',
+    [
+        (['run1'], ['run1', 'run2']),
+        (['run1', 'run11'], ['run1', 'run11', 'run2']),
+        (['run1200'], ['run1200', 'run1201']),
+        (['foo'], ['foo', 'bar']),
```
Comment (MetRonnie) [inline tests/unit/test_workflow_files.py:1535]: `runN` isn't used for named (instead of numbered) runs

```suggestion
```

Comment (wxtim) [reply tests/unit/test_workflow_files.py:1535]: No, I want to check that nothing untoward happens when one uses named runs.

**Review by datamel — COMMENTED** [review]
Fast tests failing GH Actions. Looks to me like the same ones failing on https://github.com/cylc/cylc-flow/pull/4361 which was merged into master.

**Review by MetRonnie — COMMENTED** [review]
I guess there should also be a test that `runN` is not removed if cleaning a non-latest run dir

**Review by MetRonnie — APPROVED** [review]
Tested manually, all good 👍

**Pull request merged**